cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -134,13 +134,18 @@ public synchronized void create(NamespaceMeta metadata) throws Exception {
       authorizerInstantiator.get().enforce(instanceId, principal, Action.ADMIN);
     }
 
+    // store the meta first in the namespace store because namespacedlocationfactory need to look up location
+    // mapping from namespace config
```

Comment:
To know that there is no  custom mapping. It need to figure it out somehow unless we pass down this information to it from NamespaceAdmin through parameter or  something. That will require adding another get() and take like a boolean which says customMapping or not. But doing this does not look clean to me.